core/iwasm/interpreter/wasm_loader.c

```diff
@@ -2350,6 +2350,91 @@ load_start_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+#if WASM_ENABLE_CUSTOM_NAME_SECTION != 0
+static bool
+handle_name_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
+                    char *error_buf, uint32 error_buf_size)
+{
+    const uint8 *p = buf, *p_end = buf_end;
+    uint32 name_type, subsection_size;
+    uint32 previous_name_type = 0;
+    uint32 num_func_name;
+    uint32 func_index;
+    uint32 previous_func_index = ~0U;
+    uint32 func_name_len;
+    uint32 name_index;
+    int i = 0;
+
+    if (p >= p_end) {
+        set_error_buf(error_buf, error_buf_size, "unexpected end");
+        return false;
+    }
+
+    while (p < p_end) {
+        read_leb_uint32(p, p_end, name_type);
+        if (i != 0) {
+            if (name_type == previous_name_type) {
+                set_error_buf(error_buf, error_buf_size, "duplicate sub-section");
+                return false;
+            }
+            if (name_type < previous_name_type ) {
+                set_error_buf(error_buf, error_buf_size, "out-of-order sub-section");
+                return false;
+            }
+        }
+        previous_name_type = name_type;
+        read_leb_uint32(p, p_end, subsection_size);
+        CHECK_BUF(p, p_end, subsection_size);
+        switch (name_type)
+        {
```

Comment:
had better move `{` to last line
```suggestion
        switch (name_type) {
```